### PR TITLE
Introduce block-renderer to site assembler to improve performance

### PR DESCRIPTION
**Pull request by arthur791004**
#### Proposed Changes

This PR is focusing on the block-renderer package and uses it to improve the performance of the site assembler. See the simple [README.md](https://github.com/Automattic/wp-calypso/blob/a58ad832102e59f4aa8b856bfb063390d7df3c6c/packages/block-renderer/README.md) to understand how to use the block-renderer package.

pbxlJb-39t-p2 describes the idea of the block-renderer, and pbxlJb-3gc-p2 does the performance analysis

Here is a follow-up PR to handle the scroll behavior: https://github.com/Automattic/wp-calypso/pull/71527

https://user-images.githubusercontent.com/13596067/209108222-d3114764-04b4-4d73-838f-9c5f35a75288.mov

#### Testing Instructions

* Apply D95663-code to your sandbox
* Go to `/setup?siteSlug=<your_site>&flags=pattern-assembler/client-side-render`
* Click the "Continue" button until you land on the Design Picker step
* Scroll to the bottom and select the Blank Canvas CTA
* When you're in the Pattern Assembler step
  * Verify every pattern in the pattern list looks good as before
  * Verify the large preview looks good after you select any header, sections, and footers

#### Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to https://github.com/Automattic/wp-calypso/issues/67076

### CODE REVIEW

**Comment by fushar** [comment]
Hmm, is this PR working? I get broken patterns. Tested on calypso.live and localhost, in Chrome and Firefox:

image

**Comment by arthur791004** [comment]
I just deployed D95663-code. Could you pull the latest trunk of wpcom and try it again? thanks!

File: packages/block-renderer/src/hooks/use-query-rendered-patterns.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	} );
+
+	return wpcomRequest( {
+		apiNamespace: 'wpcom/v2',
+		path: `/sites/${ encodeURIComponent( siteId ) }/block-renderer/patterns/render`,
+		query: params.toString(),
+	} );
+};
+
+const useQueryRenderedPatterns = (
```
Comment (fushar) [inline packages/block-renderer/src/hooks/use-query-rendered-patterns.ts]: The filename, function, and usage of this hook are not consistent. Can we unify them?

- `use-rendered-patterns.ts`
- `const useRenderedPatterns`
- `import useRenderedPatterns`

File: packages/block-renderer/src/hooks/use-query-settings.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { useQuery, UseQueryResult, UseQueryOptions } from 'react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+
+const useQueryBlockSettings = (
```
Comment (fushar) [inline packages/block-renderer/src/hooks/use-query-settings.ts]: Similarly here

- `use-block-settings.ts`
- `const useBlockSettings`
- `import useBlockSettings`

Comment (arthur791004) [reply packages/block-renderer/src/hooks/use-query-settings.ts]: Okay, I rename it to `useBlockRendererSetttings` 👍

**Comment by fushar** [comment]
Looks like I found some broken CSS... for example this header:

image

This PR somehow adds underlines to the menu items at the upper-right corner:

|Production|This PR|
|-|-|
|image|image|

Is this intended?

**Comment by arthur791004** [comment]
hmmm....it's weird. I can see the underline in the site editor but I cannot see it when visiting the site 😂

| Site editor | Home |
| - | - |
| image | image |

The editor styles override the `text-decoration` so I expect the behavior is the same as the site editor.

image

**Comment by fushar** [comment]
Wait really? Here is Site Editor for me; it does not have the underlines

image

**Comment by taipeicoder** [comment]
Is it related to https://github.com/Automattic/themes/issues/6541?

**Comment by arthur791004** [comment]
> Wait really? Here is Site Editor for me; it does not have the underlines

@fushar Could you help to check the styles? I go to the `dotcompatterns` site and edit the post, it also shows the underline 😂

Looks like this issue depends on the order of CSS...

**Comment by fushar** [comment]
Interesting... if I edit the post in dotcompatterns, I can see the underlines as well 😅 

image

**Comment by fushar** [comment]
I guess @taipeicoder is right; it's maybe related to that issue and not related to this PR. Let me re-review this PR again.

**Comment by arthur791004** [comment]
I'll confirm what's the expected behavior with @alaczek next week

**Review by fushar — APPROVED** [review]
Working perfectly!!

**Comment by bangank36** [comment]
> Is it related to [Automattic/themes#6541](https://github.com/Automattic/themes/issues/6541)?

This issue is reported for block-base child theme, while blank-canvas-3 is not right?

**Comment by arthur791004** [comment]
> This issue is reported for block-base child theme, while blank-canvas-3 is not right?

I'm not sure but maybe the root cause is the same

**Pull request merged**